models/user.js

```diff
@@ -89,6 +89,18 @@ UserSchema.methods.hashPassword = function(password) {
 };
 
 /**
+ * Create instance method for changing a password 
+ */
+ UserSchema.methods.changePassword = function(newPassword) {
+ 	if (newPassword && newPassword.length > 6) {
+ 		this.password = newPassword;
+ 		this.save();
```

Comment:
this.save() should be called in the controller not the model with a callback that catches the errors (like incorrect length from the validator). Check out http://mongoosejs.com/docs/api.html#model_Model-save